api/v1beta1/kafkacluster_types.go

```diff
@@ -364,6 +365,16 @@ func (c ExternalListenerConfig) GetAnyCastPort() int32 {
 	return *c.AnyCastPort
 }
 
+// When TLS is enabled AnyCastPort is enough since hostname based multiplexing
+// is used and not port based one
+func (c ExternalListenerConfig) GetBrokerPort(brokerId int32) int32 {
+	if c.TLSEnabled() {
```

Comment:
We should come up with a generic solution for generating port numbers for external listeners instead regardless wether TLS is enabled or not which allows the use of a single port.